TickSpec/ServiceProvider.fs

```diff
@@ -3,21 +3,94 @@
 open System
 open System.Collections.Generic
 open System.Diagnostics
+open System.Reflection
 
+/// Provides an instance provider for tests
+type IInstanceProvider =
+    inherit IDisposable
+
+    /// Resolves an instance of desired type
+    abstract member Resolve : Type -> obj

```

Comment:
I think IServiceProvider makes sense to use as a base interface for this and leave the name as it was - it has clear documented semantics